carma_wm/src/CARMAWorldModel.cpp

```diff
@@ -1419,6 +1429,18 @@ namespace carma_wm
 
   boost::posix_time::ptime CARMAWorldModel::min_end_time_converter_minute_of_year(boost::posix_time::ptime min_end_time,bool moy_exists,uint32_t moy, bool is_simulation)
   {
+    double simulation_time_difference_in_seconds = 0.0;
+
+    // NOTE: In simulation, ROS1 clock (often coming from CARLA) can have a large time ahead.
+    // the timing calculated here is in Simulation time, which is behind. Therefore, the world model adds the offset to make it meaningful to carma-platform:
+    // https://github.com/usdot-fhwa-stol/carma-platform/issues/2217
+    if (is_simulation && ros1_clock_.has_value() && simulation_clock_.has_value())
```

Comment:
`std::optional` is implicitly convertible to `bool`, so we don't need the `.has_value()` calls. Being explicit in this context also seems to go against the ISO C++ code guidelines on verbosity (see [here](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es87-dont-add-redundant--or--to-conditions)). Consider replacing with
```cpp
if (is_simulation && ros1_clock_ && simulation_clock_)
```